### DISCUSSION TITLE
Mock POST or PUT with empty response

### DISCUSSION THREAD

**Discussion by dontfreakout** [microcks/microcks#682] [Q&A] [answered]
Hello, 

is it possible to mock an empty POST or PUT response on Microcks 1.6.x? 
Or generally, any endpoint with empty response.

For example with this endpoint:
```yaml
/users:
  post:
      summary: Create a user
      operationId: createUser
      responses:
        '201':
          description: Created
          headers:
            Location:
              schema:
                type: string
              description: Url of the newly created user resource
      description: It creates a new user
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/User'
            examples:
              user:
                value:
                  name: John Doe
                  loginType: basic
                  password: mySecretPassword
```
I would like to have a working endpoint in Microcks, but I don't see the Mock URL available for this type of request.

I tried **x-microcks-operation**, but with no success. 
```yaml
x-microcks-operation:
        dispatcher: SCRIPT
        dispatcherRules: |
          return "Accepted"
```

Thanks for any advice

**Comment by lbroudoux** [comment, marked as answer]
Hey! Not sure if we're currently able to do that...
I think it has nothing to do with the dispatcher but with the way we gather example fragments to build a request/response pair.

Have you tried to add - at least - an empty response example with `user` has a key? Something like:
```yaml
responses:
  '201':
    content:
       application/json:
          examples:
            user:
              value: 
```
Maybe it could do the trick to allow Microcks to assemble a valid request/response pair?

**Reply by dontfreakout** [reply]
Yes, this did the trick :+1: 

Thank you for such quick help :slightly_smiling_face:

**Reply by felipe-alves-moraes** [reply]
Hi,
I tried to do the same on my exploration on Microcks, but my tests are failing complaining that the response from my service is not returning a Content-type header, which doesn't make much sense if my service does not return any answer.

Are there any way to disable the Header check for a specific test?

Microcks Version: quay.io/microcks/microcks:1.8.0-fix-1

**Reply by lbroudoux** [reply]
I think this enhancement proposal (see https://github.com/microcks/microcks/issues/944) could to the trick but I didn't find time to review it in-depth... Would you review and join the conversation here?
